src/data_structures/two_pointer.md

```diff
@@ -0,0 +1,80 @@
+# Two Pointer Algorithm
+
+## Overview
+
+The Two Pointer Algorithm is a technique used for efficiently solving problems with linear data structures like arrays or linked lists. It involves using two pointers to traverse the data structure and perform operations based on certain conditions.
+
+## Basic Idea
+
+The basic idea behind the Two Pointer Algorithm is to maintain two pointers that traverse the data structure in a way that reduces the time complexity. The pointers move towards each other or in a specific pattern based on the problem requirements.
```

Comment:
Probably want to mention that two pointers doesn't overlap, and hence it is at most $O(n)$ to traverse the whole array.